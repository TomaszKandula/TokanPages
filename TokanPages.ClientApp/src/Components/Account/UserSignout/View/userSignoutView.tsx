import * as React from "react";
import { ViewProperties } from "../../../../Shared/Abstractions";
import { CustomCard, Icon } from "../../../../Shared/Components";
import { UserSignoutProps } from "../userSignout";

interface UserSignoutViewProps extends ViewProperties, UserSignoutProps {
    languageId: string;
    caption: string;
    status: string[];
    buttonText: string;
    isAnonymous: boolean;
}

export const UserSignoutView = (props: UserSignoutViewProps): React.ReactElement => (
        <section className={props.className}>
            <div className="bulma-container bulma-is-max-tablet">
                <div className="py-6">
                    <CustomCard
                        isLoading={props.isLoading}
                        caption={props.caption}
                        text={props.status}
                        icon={<Icon name="Check" size={3} />}
                        colour="has-text-success"
                    />
                </div>
            </div>
        </section>
    );
