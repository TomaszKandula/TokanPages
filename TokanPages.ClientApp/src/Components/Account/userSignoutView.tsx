import * as React from "react";
import Container from "@material-ui/core/Container";
import Box from "@material-ui/core/Box";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import TextField from "@material-ui/core/TextField";
import { AccountCircle } from "@material-ui/icons";
import UserSignoutStyle from "./Styles/userSignoutStyle";

interface IBinding 
{
    bind: IProperties;
}

interface IProperties
{
    caption: string;
    status: string;
}

const UserSignoutView = (props: IBinding) => 
{
    const classes = UserSignoutStyle();
    return (
        <section>
            <Container maxWidth="sm">
                <Box pt={18} pb={10}>             
                    <Card elevation={4}>
                        <CardContent className={classes.card}>
                            <Box mb={3} textAlign="center">
                                <AccountCircle color="primary" style={{ fontSize: 72 }} />
                                <Typography variant="h5" component="h2" color="textSecondary">
                                    {props.bind.caption}
                                </Typography>
                            </Box>
                            <Box>
                                <Grid container spacing={2}>
                                    <Grid item xs={12}>
                                        <TextField 
                                            variant="outlined" required fullWidth 
                                            name="email" id="email" label="Email address" autoComplete="email" 
                                        />
                                    </Grid>
                                </Grid>
                                <Box my={2}>
                                    {props.bind.status}
                                </Box>
                            </Box>
                        </CardContent>   
                    </Card>                    
                </Box>
            </Container>
        </section>
    );
}

export default UserSignoutView;
