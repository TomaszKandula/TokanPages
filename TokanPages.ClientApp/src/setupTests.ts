// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
