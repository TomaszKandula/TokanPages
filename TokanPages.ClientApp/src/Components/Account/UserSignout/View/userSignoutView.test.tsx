import "../../../../setupTests";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { UserSignoutView } from "./userSignoutView";

describe("test account group component: userSignoutView", () => {
    it("should render correctly '<UserSignoutView />' when content is loaded.", () => {
        const html = render(
            <BrowserRouter>
                <UserSignoutView
                    isLoading={false}
                    languageId="en"
                    caption="Sign out"
                    status="Signing out current user..., please wait."
                    buttonText={"Go back to main"}
                    isAnonymous={false}
                    pt={15}
                    pb={15}
                    background={{ backgroundColor: "white" }}
                />
            </BrowserRouter>
        );

        expect(html).toMatchSnapshot();
    });
});
