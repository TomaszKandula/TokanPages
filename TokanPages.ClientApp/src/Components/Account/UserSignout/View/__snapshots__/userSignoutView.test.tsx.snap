// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test account group component: userSignoutView should render correctly '<UserSignoutView />' when content is loaded. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="makeStyles-section-1"
        style="background-color: white;"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm"
        >
          <div
            class="MuiBox-root MuiBox-root-9"
          >
            <div
              class="MuiPaper-root MuiCard-root makeStyles-card-5 MuiPaper-elevation0 MuiPaper-rounded"
            >
              <div
                class="MuiCardContent-root makeStyles-card_content-6"
              >
                <div
                  class="MuiBox-root MuiBox-root-10"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-account-2"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                    />
                  </svg>
                  <p
                    class="MuiTypography-root makeStyles-caption-3 MuiTypography-body1"
                  >
                    Sign out
                  </p>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-11"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <p
                        class="MuiTypography-root makeStyles-status-4 MuiTypography-body1"
                      >
                        Signing out current user..., please wait.
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-12"
                >
                  <a
                    class="makeStyles-link-7"
                    href="/"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-8 Mui-disabled MuiButton-fullWidth Mui-disabled"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Go back to main
                      </span>
                    </button>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="makeStyles-section-1"
      style="background-color: white;"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm"
      >
        <div
          class="MuiBox-root MuiBox-root-9"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-card-5 MuiPaper-elevation0 MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root makeStyles-card_content-6"
            >
              <div
                class="MuiBox-root MuiBox-root-10"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-account-2"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                  />
                </svg>
                <p
                  class="MuiTypography-root makeStyles-caption-3 MuiTypography-body1"
                >
                  Sign out
                </p>
              </div>
              <div
                class="MuiBox-root MuiBox-root-11"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  >
                    <p
                      class="MuiTypography-root makeStyles-status-4 MuiTypography-body1"
                    >
                      Signing out current user..., please wait.
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root MuiBox-root-12"
              >
                <a
                  class="makeStyles-link-7"
                  href="/"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-8 Mui-disabled MuiButton-fullWidth Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Go back to main
                    </span>
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
